src/modules/decklink/consumer_decklink.cpp

```diff
@@ -599,6 +587,9 @@ class DeckLinkConsumer
 
 					if ( !progressive && m_displayMode->GetFieldDominance() == bmdUpperFieldFirst )
 					{
+						// Make the first line black for field order correction.
```

Comment:
I agree with Dan's comments about using "top_field_first" property to determine the mlt frame field order.

Another way to correct the field order would be to delay the output by one field. Display the upper field from this frame and the lower field from the next frame.